server/src/graql/gremlin/TraversalPlanner.java

```diff
@@ -207,15 +217,17 @@ public static GraqlTraversal createTraversal(Pattern pattern, TransactionOLTP tx
         return weightedGraph;
     }
 
-    private static Set<Node> chooseStartingNodeSet(Set<Fragment> fragmentSet, Map<NodeId, Node> allNodes, SparseWeightedGraph sparseWeightedGraph) {
+    private static Set<Node> chooseStartingNodeSet(Set<Fragment> fragmentSet, Map<NodeId, Node> allNodes, SparseWeightedGraph sparseWeightedGraph, TransactionOLTP tx) {
         final Set<Node> highPriorityStartingNodeSet = new HashSet<>();
 
-        fragmentSet.forEach(fragment -> {
-            if (fragment.hasFixedFragmentCost()) {
+        fragmentSet.stream()
+            .filter(Fragment::hasFixedFragmentCost)
+            .sorted(Comparator.comparing(fragment -> estimateInitialCost(fragment, tx)))
+            .limit(3)
```

Comment:
maybe let's record it as a variable with a meaningful name instead of a magic number. Will help in the future.

Reply:
changed to MAX_STARTING_POINTS